src/change_stream/mod.rs

```diff
@@ -0,0 +1,170 @@
+//! Contains the functionality for ChangeStreams.
+pub mod document;
+pub mod option;
+
+use std::{marker::PhantomData, pin::Pin, task::Poll};
+
+use bson::Document;
+use futures::{stream::Stream, task::Context};
+use serde::{de::DeserializeOwned, Deserialize};
+
+use self::document::*;
+use crate::{
+    change_stream::option::ChangeStreamOptions,
+    error::Result,
+    options::AggregateOptions,
+    selection_criteria::ReadPreference,
+    Client,
+    Collection,
+    Cursor,
+    Database,
+};
+
+/// A `ChangeStream` streams the ongoing changes of its associated collection, database or
+/// deployment. `ChangeStream` instances should be created with method `watch` against the relevant
+/// target. Issuing a raw change stream aggregation is discouraged unless users wish to explicitly
+/// opt out of resumability.
+///
+/// A `ChangeStream` can be iterated to return batches of instances of any type that implements the
+/// Deserialize trait. By default, it returns batches of `ChangeStreamEventDocument`. These
+/// documents correspond to changes in the associated collection, database or deployment.
+///
+/// A change stream can be iterated like any other `Iterator`:
+///
+/// ```rust
+/// #[cfg(not(feature = "sync"))]
+/// # use futures::stream::StreamExt;
+
+/// # use mongodb::{Client, error::Result};
+/// # #[cfg(feature = "async-std-runtime")]
+/// # use async_std::task;
+/// # #[cfg(feature = "tokio-runtime")]
+/// # use tokio::task;
+/// #
+/// # #[cfg(not(feature = "sync"))]
+/// # async fn func() -> Result<()> {
+/// # let client = Client::with_uri_str("mongodb://example.com").await?;
+/// # let coll = client.database("foo").collection("bar");
+/// let mut change_stream = coll.watch(None, None).await?;
+/// while let Some(change) = change_stream.next().await {
+///     println!("{:?}", change?)
+/// }
+/// #
+/// # Ok(())
+/// # }
+/// ```
+/// 
+/// A change stream allows any Deserialize type for `ChangeStreamEventDocument` to be specified:
+/// ```rust
+/// #[cfg(not(feature = "sync"))]
+/// # use bson::Document;
+/// # use futures::stream::StreamExt;
+/// # use serde::Deserialize;
+
+/// # use mongodb::{Client, error::Result};
+/// # #[cfg(feature = "async-std-runtime")]
+/// # use async_std::task;
+/// # #[cfg(feature = "tokio-runtime")]
+/// # use tokio::task;
+/// #[derive(Deserialize, Debug)]
+/// pub struct EventDocument {
+///     pub full_document: Option<Document>,
+/// }
+/// #
+/// # #[cfg(not(feature = "sync"))]
+/// # async fn func() -> Result<()> {
+/// # let client = Client::with_uri_str("mongodb://example.com").await?;
+/// # let coll = client.database("foo").collection("bar");
+/// let mut change_stream = coll.watch(None, None).await?;
+/// while let Some(change) = change_stream.next().await {
+///     let change: EventDocument = change?;
+///     println!("{:?}", change)
+/// }
+/// #
+/// # Ok(())
+/// # }
+/// ```
+
```

Comment:
these docstrings need to be connected I believe.

Reply:
fixed